File: core/src/main/java/heroes/journey/registries/Registries.java
package heroes.journey.registries;

import heroes.journey.entities.Buff;
import heroes.journey.entities.Challenge;
import heroes.journey.entities.Quest;
import heroes.journey.entities.actions.Action;
import heroes.journey.entities.items.Item;
import heroes.journey.entities.items.ItemSubType;
import heroes.journey.tilemap.Biome;
import heroes.journey.tilemap.FeatureType;
import heroes.journey.tilemap.TileBatch;
import heroes.journey.tilemap.TileLayout;
import heroes.journey.tilemap.wavefunctiontiles.Terrain;

public class Registries {

    public static Registry<Item> ItemManager = new Registry<>();
    public static Registry<ItemSubType> ItemSubTypeManager = new Registry<>();
    
    public static Registry<Action> ActionManager = new Registry<>();
    public static Registry<Buff> BuffManager = new Registry<>();

    public static Registry<Quest> QuestManager = new Registry<>();
    public static Registry<Challenge> ChallengeManager = new Registry<>();

    public static Registry<Biome> BiomeManager = new Registry<>();
    public static Registry<FeatureType> FeatureTypeManager = new Registry<>();

    public static Registry<Terrain> TerrainManager = new Registry<>();
    public static Registry<TileLayout> TileLayoutManager = new Registry<>();
    public static Registry<TileBatch> TileBatchManager = new Registry<>();

}

File: modlib/src/main/java/heroes/journey/modlib/Registries.kt
package heroes.journey.modlib

import heroes.journey.modlib.actions.IAction

object Registries {
    lateinit var QuestManager: Registry<IQuest>
    lateinit var StatManager: Registry<IStat>
    lateinit var BuffManager: Registry<IBuff>
    lateinit var ChallengeManager: Registry<IChallenge>
    lateinit var ActionManager: Registry<IAction>
    lateinit var ItemManager: Registry<IItem>
    lateinit var ItemSubTypeManager: Registry<IItemSubType>
    lateinit var BiomeManager: Registry<IBiome>
    lateinit var FeatureTypeManager: Registry<IFeatureType>
    lateinit var TerrainManager: Registry<ITerrain>
    lateinit var TileLayoutManager: Registry<ITileLayout>
    lateinit var TileBatchManager: Registry<ITileBatch>
    lateinit var GroupManager: Registry<IGroup>
    lateinit var RenderableManager: Registry<IRenderable>
    lateinit var TextureManager: Registry<ITextureMap>
}
